### DISCUSSION TITLE
.NET 9 vs .NET 8

### DISCUSSION THREAD

**Discussion by GalaxiaGuy** [dotnet/maui#21386] [General]
It seems a lot of fixes are going into .NET 8 service releases. In there any guidance on what will go into .NET 9 that _won't_ make it into .NET 8?

**Comment by RobTF** [comment]
In terms of bug fixes it had better be nothing until .NET 8 is out of support. The .NET 7 to 8 transition was abysmal - there is no excuse with .NET 8 as it's an LTS release. Using the fact .NET 7 was an STS release to explain why fixes weren't being backported was relied upon **heavily** by the MAUI team.

**Reply by kcrg** [reply]
MAUI does not have any LTS or STS releases. **All** versions are supported for 1.5 years.

https://dotnet.microsoft.com/en-us/platform/support/policy/maui

**Reply by RobTF** [reply]
@kcrg you are correct and I forgot about that, however 1.5 years is in line with the support duration of an STS .NET release and .NET 7 was not out of support before fixes were coming out that it did not receive in favour of the yet to be released .NET 8. This put developers in a position whereby they had to use the supported framework with MAUI bugs, or use a preview/release candidate version of .NET to get MAUI support. It was poor and I really hope we don't get a repeat of this.

This problem will be worse for an LTS release as you could update to .NET 8 and half way through its support lifecycle MAUI support runs out and you're forced onto .NET 9 prematurely. I doubt it will be possible to run MAUI 9 on .NET 8, which makes the .NET 8/LTS support story worthless for anyone using the MAUI libraries.

If MAUI has different support requirements etc. I don't get why they align with .NET at all? Just disconnect entirely and release how and when required. My team and I wouldn't care if MAUI 13 came out midway through the .NET 10 release with its 1.5 years support as we would just deal with it differently like we would Newtonsoft or Automapper. The whole MAUI 6/7/8 stuff and forcing releases in line with .NET feels more like a branding exercise as opposed to anything which helps developers.
